financial/exponential_moving_average.py

```diff
@@ -0,0 +1,70 @@
+"""
+    Calculates exponential moving average (EMA) on the series of numbers
+    Wikipedia Reference: https://en.wikipedia.org/wiki/Exponential_smoothing
+    Reference: https://www.investopedia.com/terms/e/ema.asp#toc-what-is-an-exponential-moving-average-ema
+
+    Exponential moving average is used in finance to analyze changes stock prices.
+    EMA is used in conjunction with Simple moving average (SMA), EMA reacts to the
+    changes in the value quicker than SMA, which is one of the advantages of using EMA.
+"""
+
+
+def exponential_moving_average(series: list[float], window_size: int) -> list[float]:
+    """
+    Returns the exponential moving average of the given array list
+    >>> exponential_moving_average([2, 5, 3, 8.2, 6, 9, 10], 3)
```

Comment:
```suggestion
The financial calculations seem artificial and trivial when they are supplied with all the data upfront.

Could we convert `series: list[float]` --> `series: Iterable[float]` so that we could continuously create/add to an EMA as new data came in from some generator expression?  We will no longer be able to do `len(series)` but will other issues arise?

```suggestion
from collections.abc import Iterable


def exponential_moving_average(series: Iteratable[float], window_size: int) -> list[float]:
    """
    Returns the exponential moving average of the given array list
    >>> exponential_moving_average(iter([2, 5, 3, 8.2, 6, 9, 10]), 3)
```

Reply:
Instead of `return list[float]` could we consider yielding floats as we finish windows?

Reply:
Thanks for the suggestion!
I have added generator as parameter instead of an iterator which acts as mock for file or data stream.